caliperreader: fix duplicate dataframe rows

slabasan opened the pull request:
root nids were mapping to multiple series of metrics, this merges the metrics for the same node into a single dataframe row

slabasan commented:
This PR ready for review.

slabasan commented on hatchet/tests/callgrind.py:
```diff
@@ -23,7 +25,11 @@ def test_graphframe(calc_pi_callgrind_dot):
+        if col == "time (inc)" or col == "time":
+            assert gf.dataframe[col].dtype == np.float64
+        elif col == "name" or col == "module" or col == "node":
+            assert gf.dataframe[col].dtype == np.object
```
```python
if col in ("time (inc)", "time"):
    #assert here
elif col in ("name", "module", "node"):
    #assert here
```

slabasan commented on hatchet/readers/caliper_reader.py:
```diff
@@ -305,6 +306,8 @@ def read(self):
                 indices.append("rank")
             dataframe.set_index(indices, drop=False, inplace=True)
 
+        dataframe["nid"] = dataframe["nid"].astype(np.int64)
```
The `data` field in the `lulesh-sample-annotation-profile.json` includes `null`/`None` values for the `path` column, forcing it to be a float type. Why does the `path` column include `null`?

```json
       [ 21948.000000, 0, 21948.000000, null ],
        [ 23349.000000, 1, 23349.000000, null ],
        [ 22185.000000, 2, 22185.000000, null ],
        [ 22366.000000, 3, 22366.000000, null ],
        [ 7980.000000, 4, 7980.000000, null ],
        [ 7205.000000, 5, 7205.000000, null ],
        [ 7579.000000, 6, 7579.000000, null ],
        [ 15513.000000, 7, 15513.000000, null ],
        [ 5882425.000000, 0, 121489.000000, 0 ],
        [ 5902349.000000, 1, 105528.000000, 0 ],
        [ 5898577.000000, 2, 110799.000000, 0 ],
        [ 5882996.000000, 3, 113830.000000, 0 ],
        [ 5905595.000000, 4, 118953.000000, 0 ],
        [ 5877613.000000, 5, 133256.000000, 0 ],
        [ 5870933.000000, 6, 114035.000000, 0 ],
        [ 5898724.000000, 7, 137098.000000, 0 ],
```

bhatele commented on hatchet/readers/caliper_reader.py:
```diff
@@ -74,6 +74,16 @@ def read_json_sections(self):
+        # remove data entries containing None (null in json file)
+        # first, parse json_data list of lists to identify lists containing None
+        entries_to_remove = []
+        for sublist in self.json_data:
```
It'll be more precise if we just did this for the path column in case some other columns have a None in some rows.

slabasan committed:
caliperreader: fix duplicate dataframe rows

root nids were mapping to multiple series of metrics, this merges the metrics
for the same node into a single dataframe row
